### PR TITLE
Custom gate view

### PR DESCRIPTION
**Pull request by JonathanAndrews**
- Users can run `rails generate balrog:view` and copy the balrog gate view into their own app.
- Once in their own app, they can style and modify the gate.

### CODE REVIEW

File: lib/balrog/generators/view_generator.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  desc "Copies the Balrog gate view and layout into your application, where you can edit and style them."
+
+  def copy_gate_view
+    gate_view = File.open(
+      File.join(__dir__, '../../../', 'app/views/balrog/gate.html.erb'),
+      'r')
+
+    content = gate_view.read
+    gate_view.close
```
Comment (em-cd) [inline lib/balrog/generators/view_generator.rb]: I'm not too sure about this, but wouldn't you want to close the file *after* you've used the content on the next line?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# 1.1.0
```
Comment (jalada) [inline CHANGELOG.md]: In future I suggest only updating the CHANGELOG on master, e.g. once things are merged. It'll make conflicts a little easier to resolve.

File: lib/balrog/generators/view_generator.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Balrog::ViewGenerator < Rails::Generators::Base
+
+  desc "Copies the Balrog gate view and layout into your application, where you can edit and style them."
+
+  def copy_gate_view
+    gate_view = File.open(
+      File.join(__dir__, '../../../', 'app/views/balrog/gate.html.erb'),
```
Comment (jalada) [inline lib/balrog/generators/view_generator.rb]: I don't think this style works on Windows, but I'm not sure. Did you adapt this from somewhere else?

File: lib/balrog/generators/view_generator.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      'r')
+
+    content = gate_view.read
+    gate_view.close
+
+    create_file "app/views/balrog/gate.html.erb", content
+  end
+
+  def copy_layout
```
[the remark this answers is not in the document]
Comment (JonathanAndrews) [reply lib/balrog/generators/view_generator.rb]: I split the balrog view up into layout and view. 
That way, once users have imported the views, they can import stylesheets in the balrog layout (spec/dummy-rails-app/app/views/layouts/balrog.html.erb) and not directly on a view that contains the balrog form.

File: lib/balrog/generators/view_generator.rb (hunk 3 above)
Comment (JonathanAndrews) [reply lib/balrog/generators/view_generator.rb]: [Devise](https://github.com/plataformatec/devise/blob/14863ba4c92cd9781a961be0486f0ea7dfe84144/lib/generators/devise/views_generator.rb#L75) do a similar thing.

Comment (jalada) [reply lib/balrog/generators/view_generator.rb]: Thx

File: lib/balrog/generators/view_generator.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (jalada) [reply lib/balrog/generators/view_generator.rb]: Gotcha. Sorry, my lack of understanding of Rails generators, I was hunting for where it was called. Rails just automatically calls both methods in here if you run the generator?

**Comment by jalada** [comment]
:shipit: !

File: lib/balrog/generators/view_generator.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (JonathanAndrews) [reply lib/balrog/generators/view_generator.rb]: Yep 👍

**Pull request merged**